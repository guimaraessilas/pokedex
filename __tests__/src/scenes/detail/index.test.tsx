import { render, RenderResult } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import Detail from '../../../../src/scenes/detail';
import { store } from '../../../../store';

jest.mock('@expo/vector-icons', () => 'Icon');
jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
const mockedFunction = jest.fn();

const props = {
  navigation: {
    goBack: mockedFunction,
  },
};

const renderDetailScreen = (): RenderResult =>
  render(
    <Provider store={store}>
      <Detail {...props} />
    </Provider>
  );

describe('Should test detail screen', () => {
  describe('test snapshot', () => {
    it('should render detail screen correctly', () => {
      const tree = renderDetailScreen().toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
