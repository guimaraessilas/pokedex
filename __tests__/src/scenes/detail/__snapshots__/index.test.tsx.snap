// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test detail screen test snapshot should render detail screen correctly 1`] = `null`;
